src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -313,6 +324,40 @@ static int wall_tty_block(void) {
         return fd;
 }
 
+static int parse_filter(const char *arg) {
+        const char *l;
+        bool strict = false;
+        int r;
+        pid_t p;
+
+        assert(arg || !strict);
+
+        if (!arg)
+                goto default_noarg;
+
+        if (streq(arg, "all")) {
+                arg_query_pid = 0;
+                return 1;
+        }
+
+        l = startswith(arg, "pid="); // IDEA: Support other filter values
```

Comment:
i don't think we want two levels of switches. just introduce separate --filter-foo= or --filter-bar= 1st level switches if needed.